lib/view/community/detailed_thread_view.dart

```diff
@@ -1,211 +1,275 @@
-                      .style
-                      .apply(fontSizeFactor: 1.8, fontWeightDelta: 2)),
-            ),
-            Row(
-              mainAxisAlignment: MainAxisAlignment.start,
-              children: <Widget>[
-                const SizedBox(width: 16),
-                Text(widget.thread.author.username,
-                    style: DefaultTextStyle.of(context).style.apply(
-                        color: Colors.grey[700], fontStyle: FontStyle.italic)),
-                const SizedBox(width: 16),
-                Text(
-                    DateFormat('yyyy-MM-dd kk:mm')
-                        .format(widget.thread.timestamp),
-                    style: DefaultTextStyle.of(context).style.apply(
-                        color: Colors.grey[700], fontStyle: FontStyle.italic)),
-              ],
-            ),
-            ListTile(
-              title: Text(widget.thread.content, softWrap: true),
-            ),
-            // const SizedBox(height: 8),
-            Row(
-              // mainAxisAlignment: MainAxisAlignment.end,
-              children: <Widget>[
-                const SizedBox(width: 16),
-                LikeButton(
-                  isLiked: widget.thread.isLikedBy(widget.currUser),
-                  likeCount: widget.thread.numLikes(),
-                  onTap: _handleTapLike,
-                  countPostion: CountPostion.left,
-                  circleColor: const CircleColor(
-                      start: Colors.cyanAccent, end: Colors.cyan),
-                  bubblesColor: const BubblesColor(
-                    dotPrimaryColor: Colors.lightBlue,
-                    dotSecondaryColor: Colors.blueAccent,
+    return Consumer2<Thread_Manager, User_Manager>(
+        builder: (context, threadMgr, userMgr, child) {
+          return Center(
+            child: Card(
+              child: Column(
+                mainAxisSize: MainAxisSize.min,
+                children: <Widget>[
+                  ListTile(
+                    title: Text(threadMgr.getThreadByID(widget.threadID)!.title ?? "Untitled Thread",
+                        style: DefaultTextStyle
+                            .of(context)
+                            .style
+                            .apply(fontSizeFactor: 1.8, fontWeightDelta: 2)),
+                    trailing: (widget.currUserID == threadMgr.getThreadByID(widget.threadID)!.id)
+                        ? PopupMenuButton(
+                        onSelected: (value) {
+                          setState(() {
+                            if (value == "edit") {
+                              // TODO ROUTING: navigate to edit thread screen
+                            } else if (value == "delete") {
+                              threadMgr.removeThread(widget.threadID);
+                              // TODO ROUTING: fix this. screen turns red bc thread is delated. route to threadlist view?
+                            }
+                          });
```

Comment:
need some help to add routing to editThread after the user presses Edit thread, and routing back to thread list view after the user presses Delete thread

Reply:
@RealDyllon @celerie could you assist? thank you very much